### DISCUSSION TITLE
Animated.Flatlist is Not a Pure Component if i give him style prop?

### DISCUSSION THREAD

**Discussion by gaganIcodelabs** [software-mansion/react-native-reanimated#6426] [Q&A]
So I have been experimenting with Animated.Flatlist .... I encountered a case that if i give him style prop even a global static object. It will no longer act as PureComponent i.e. it will rerender all its items even if all the props values remain same in next rerender.
Has anyone experienced this ????
(This does not happen with normal react-native Flatlist component)

**Comment by MatiPl01** [comment]
Hey!
Could you please provide our code? I think you are doing something wrong. I haven't noticed any difference when I added the `style` property to the `Animated.FlatList`.

Do you wrap your list items with React `memo` function? FlatList re-renders items while you scroll multiple times and it is not related to reanimated. Reanimated doesn't change how `FlatList` works except for adding a few features.
